core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParser.scala

```diff
@@ -201,45 +221,47 @@ trait SlickExpressionParser extends JavaTokenParsers {
       workspace <- workspaceQuery.findByIdQuery(context.workspaceContext.workspaceId)
       attribute <- workspaceAttributeQuery if attribute.ownerId === workspace.id && attribute.name === attrName.name && attribute.namespace === attrName.namespace
       nextEntity <- entityQuery if attribute.valueEntityRef === nextEntity.id
-    } yield (rootEntity.name, nextEntity)
+    } yield (rootEntity.name, nextEntity, attribute.listLength.isDefined)
   }
 
   // add pipe to an entity referenced by the current entity
   private def entityNameAttributePipeFunc(attrName: AttributeName)(context: SlickExpressionContext, queryPipeline: PipeType): PipeType = {
     (for {
-      (rootEntityName, entity) <- queryPipeline
+      (rootEntityName, entity, haveExplodedEntityList) <- queryPipeline
       attribute <- entityAttributeQuery if entity.id === attribute.ownerId && attribute.name === attrName.name && attribute.namespace === attrName.namespace
       nextEntity <- entityQuery if attribute.valueEntityRef === nextEntity.id
-    } yield (rootEntityName, nextEntity)).sortBy({case (nm, ent) => ent.name })
+    } yield {
+      (rootEntityName, nextEntity, haveExplodedEntityList || attribute.listLength.isDefined)
```

Comment:
Part 2 of where the magic happens. As we traverse entity refs, update the flag if any of them are ever list elements.